### PR TITLE
Issue/4657 add saturation promotion to food skills

### PR DESCRIPTION
**Pull request by Lyther**
Currently, food skills only affect hunger but not saturation. I've added the support to increase saturation from food skills.

## Trade-off with Spigot API

Spigot API only has methods to get and set saturation for a player. I have to make a saturation table of each food. Later, if spigot API adds more methods about saturation, I'll refactor my code.

By the way, the food in 1.17 is in the `Hacky support for 1.17` codes, which can run under 1.17 without error.

### CODE REVIEW

**Comment by dodieboy** [comment]
fix #4657

**Comment by nossr50** [comment]
I'll try to get this reviewed by Monday

**Comment by Chew** [comment]
Make sure to include "Fixes #x" in your issue description so it closes if this is merge :)

File: src/main/java/com/gmail/nossr50/util/skills/SkillUtils.java [hunk 1]
```diff
@@ -90,6 +90,18 @@ public static int handleFoodSkills(Player player, int eventFoodLevel, SubSkillTy
...
+    //Saturation maximum is 10 instead of 20
+    public static float handleFoodSkillsSaturation(Player player, float eventSaturationLevel, SubSkillType subSkillType) {
+        int curRank = RankUtils.getRank(player, subSkillType);
+
+        float currentSaturationLevel = player.getSaturation();
+        float saturationChange = eventSaturationLevel - currentSaturationLevel;
+
+        saturationChange += (curRank/2.0F);
+
+        return (currentSaturationLevel + saturationChange);
```
Comment (nossr50) [inline src/main/java/com/gmail/nossr50/util/skills/SkillUtils.java:102]: Do we need a Math.max() for this line? What happens if you go above 20?

File: src/main/java/com/gmail/nossr50/util/MaterialMapStore.java [hunk 2]
```diff
@@ -198,6 +200,49 @@ private void fillTierMap() {
         }
     }
 
+    private void fillFoodSaturation() {
+        foodSaturation.put(Material.APPLE, 1.2F);
```
Comment (nossr50) [inline src/main/java/com/gmail/nossr50/util/MaterialMapStore.java:204]: Where did you pull these values from?

File: src/main/java/com/gmail/nossr50/skills/fishing/FishingManager.java [hunk 3]
```diff
@@ -213,14 +213,25 @@ public double getShakeProbability() {
     }
 
     /**
-     * Handle the Fisherman's Diet ability
+     * Handle the Fisherman's Diet ability about hunger
```
Comment (nossr50) [inline src/main/java/com/gmail/nossr50/skills/fishing/FishingManager.java:216]: I suggest changing "about hunger" to "for hunger"

**Review by nossr50 — CHANGES_REQUESTED** [review]
Answer the questions when you get a chance, thanks for the PR.

**Pull request closed without merging**